### PR TITLE
bpo-45680: ``typing`` docs: improve links to docs on ``GenericAlias``/``__class_getitem__``

### PR DESCRIPTION
**Pull request by AlexWaygood**
This PR has been split off from #29335

https://bugs.python.org/issue45680

### CODE REVIEW

File: Doc/library/typing.rst [hunk 1]
```diff
@@ -49,8 +49,8 @@ annotations. These include:
      *Introducing* :class:`Protocol` and the
      :func:`@runtime_checkable<runtime_checkable>` decorator
 * :pep:`585`: Type Hinting Generics In Standard Collections
-     *Introducing* the ability to use builtin collections and ABCs as
-     :term:`generic types<generic type>`
+     *Introducing* :class:`types.GenericAlias` and the ability to use standard
+     library classes as :ref:`generic types<types-genericalias>`
```
Comment (Fidget-Spinner) [inline Doc/library/typing.rst:53]: Wait, why'd the link change from the one in the glossary to about `types-genericalias`?

File: Misc/NEWS.d/next/Documentation/2021-11-03-14-51-03.bpo-45680.9_NTFU.rst [hunk 2]
```diff
@@ -0,0 +1,2 @@
+Improve links to ``__class_getitem__`` and ``GenericAlias`` docs in the
+`typing` module's documentation.
```
Comment (Fidget-Spinner) [inline Misc/NEWS.d/next/Documentation/2021-11-03-14-51-03.bpo-45680.9_NTFU.rst]: FYI, you can write one single news item for all 3 PRs, I will then apply `skip news` label and merge subsequent PRs without a news.

You can use the news at https://github.com/python/cpython/pull/29335/files, then delete the news for the other two.

**Review by Fidget-Spinner — COMMENTED** [review]
Mostly looks good to me, just one question and comment.

File: Doc/library/typing.rst (hunk 1 above)
Comment (AlexWaygood) [reply Doc/library/typing.rst:53]: My reasoning was:

- I think the `types-genericalias` link is more informative about the change that was introduced in the PEP.
- I think it's an important piece of documentation that deserves a prominent mention near the top of the `typing` docs.
- It's consistent with the link to `types-union` in the bullet point about PEP 604.

But I'm very happy to revert this change if you think the link to the glossary is more appropriate here 🙂

File: Misc/NEWS.d/next/Documentation/2021-11-03-14-51-03.bpo-45680.9_NTFU.rst (hunk 2 above)
Comment (AlexWaygood) [reply Misc/NEWS.d/next/Documentation/2021-11-03-14-51-03.bpo-45680.9_NTFU.rst]: @Fidget-Spinner done!

**Review by Fidget-Spinner — APPROVED** [review]

**Pull request merged**